File: backend/app/schemas/notifications.py
from pydantic import BaseModel, Field


class NotificationItem(BaseModel):
    id: str
    userId: str
    type: str
    message: str
    isRead: bool
    reportId: str | None = None
    createdAt: str


class NotificationListResponse(BaseModel):
    notifications: list[NotificationItem]
    unreadCount: int = Field(ge=0)

File: frontend/lib/contracts/notifications.ts
export type NotificationType = "report_ready" | "error" | "info";

export interface NotificationItem {
  id: string;
  userId: string;
  type: NotificationType | string;
  message: string;
  isRead: boolean;
  reportId?: string;
  createdAt: string;
}

export interface NotificationListResponse {
  notifications: NotificationItem[];
  unreadCount: number;
}
